src/main/scala/beam/agentsim/agents/vehicles/BeamVehicle.scala

```diff
@@ -291,12 +291,12 @@ object BeamVehicle {
     beamLeg: BeamLeg,
     theVehicleType: BeamVehicleType,
     networkHelper: NetworkHelper,
-    onlyLength_Id_And_Type: Boolean = false
+    vehicleMappingExists: Boolean = false
   ): IndexedSeq[FuelConsumptionData] = {
     //TODO: This method is becoming a little clunky. If it has to grow again then maybe refactor/break it out
     if (beamLeg.mode.isTransit & !Modes.isOnStreetTransit(beamLeg.mode)) {
       Vector.empty
-    } else if (onlyLength_Id_And_Type) {
+    } else if (!vehicleMappingExists) {
```

Comment:
The reason negating this results in the correct output is because it goes back to pulling in more data than is needed. A more accurate fix would be to include the speed. I could've swore when we ran this originally it pulled this in...but I can't recall what happened and why this was turned into a None at some point. Either way, keeping the boolean check as is is correct - but providing the necessary speed is the better path.

Reply:
OK sounds good, thanks--so to clarify:

As originally written, collectFuelConsumptionData can return fuelConsumptionData in two ways: One with only link length, link ID, and vehicle type, and one with those plus things like link speed, free flow speed, and number of lanes. The first way is called when a vehicle energy rate mapping file exists, and the second is called when it doesn't. One way we could get RouteE to correctly calculate energy use is by passing speed as well when the first way is used.

I'm still a little confused because it seems to me that the first way (return just length, id, and vehicle type) is all that we would need to estimate energy using the default energy consumption in J/m, whereas the second way already includes the information (such as speed and # lanes) that also can be used in RouteE but isn't used for the default energy calculations. Free flow speed is also passed in the second way, and as far as I can see that is not used in RouteE, so that does seem like something we could get rid of to avoid passing unnecessary information.

Reply:
I believe the 2nd way is in fact going to go away sometime soon afaik, but we didn't do it yet.

Also, you are correct that lanes need to be included. I am not sure what happened that this was lost at some point...I swear it was there...or else it wouldn't have worked at first....